ecp-provider/provider.c

```diff
@@ -0,0 +1,728 @@
+// Copyright 2023 Google LLC this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+/*
+This provider implements ECP (Enterprise Certificate Provider) for OpenSSL. It
```

Comment:
nit: Overloading the ECP acronym here - how about "This OpenSSL provider is implemented using ECP (Enterprise Certificate Proxy)". Can refer to this as "ECPP" later if you'd like :D

Reply:
Rephrased a bit !